src/test/java/de/retest/recheck/report/ActionReplayResultTest.java

```diff
@@ -109,4 +111,25 @@ void getDiffrences_should_ignore_attributes_changes() {
 		assertThat( differences.getWindows() ).isEmpty();
 	}
 
+	@Test
+	void factory_method_should_return_no_difference_if_state_is_null() {
+		final ActionReplayData data = mock( ActionReplayData.class );
+		final SutState sutState = mock( SutState.class );
+		final StateDifference difference = null; // This difference should be null vs empty
+		final ActionReplayResult result =
+				ActionReplayResult.createActionReplayResult( data, null, null, difference, 0L, sutState );
+
+		assertThat( result.hasDifferences() ).isFalse();
+	}
+
+	@Test
+	void factory_method_should_return_no_difference_if_state_is_empty() {
+		final ActionReplayData data = mock( ActionReplayData.class );
+		final SutState sutState = mock( SutState.class );
+		final StateDifference difference = mock( StateDifference.class ); // This difference should be empty vs null
```

Comment:
Is this a "valid" situation? Having an empty state difference? Shouldn't it be `null` or differences, nothing in between?

Reply:
Well, it was checked there before, so I added this test.

But ideally, we would get rid of the `null`s and use empty `StateDifference` to remove the tedious `null`-checks.

Reply:
Allright, let me approve this one.